src/ansys/units/unit.py

```diff
@@ -407,10 +407,11 @@ def _units_to_dim(
         if unit_term in _base_units:
             idx = _base_units[unit_term]["type"]
 
-            if BaseDimensions[idx] in dimensions:
-                dimensions[BaseDimensions[idx]] += unit_term_exponent
-            else:
-                dimensions[BaseDimensions[idx]] = unit_term_exponent
+            if idx in list(BaseDimensions.__members__.keys()):
```

Comment:
`idx in BaseDimensions.__members__` is faster than `idx in list(BaseDimensions.__members__.keys())`. Done. Thank you.

Reply:
The fact that we access it via `__members__` is (to me) a clear signal that a better way to do it is available.

Reply:
Note that `hasattr(<enum>, key)` works. Alternatively, `<enum>[key]` could be done in a try-except block.

Reply:
@seanpearsonuk Yes, right. I have already used `hasattr` in previous commits but `hasattr` and `list(BaseDimensions.__members__.keys())` have same complexity O(n). Due to fixed and small size in our Enum case I had used `list(BaseDimensions.__members__.keys())` before but `BaseDimensions.__members__` is faster than both with O(1).

It is also recommended to use `__members__` method.

![image](https://github.com/user-attachments/assets/2374f2ec-e71a-43d7-b09e-1782b7106385)

Reply:
Regarding use of `<enum>[key]` in a try-except block, It will raise an exception and it's handling will be significantly slower than regular control flow operations. The cost of raising and catching exceptions can be much higher than simple conditional checks. Therefore I do not prefer try-except block.

Reply:
If we want to avoid private property usage then we can use `hasattr` instead of try-except block.

Reply:
We are using `hasattr` now.

Reply:
We are not handling at this level.